### PR TITLE
mmcsd: Remove the not really used capacity field

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
to avoid the check of CONFIG_HAVE_LONG_LONG in many place

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

File: drivers/mmcsd/mmcsd_sdio.c [hunk 1]
```diff
@@ -3377,8 +3359,8 @@ static int mmcsd_probe(FAR struct mmcsd_state_s *priv)
             {
               /* Yes...  */
 
-              finfo("Capacity: %lu Kbytes\n",
-                    (unsigned long)(priv->capacity / 1024));
+              finfo("Capacity: %lu blocks\n",
+                    (unsigned long)priv->nblocks);
```
Comment (pkarashchenko) [inline drivers/mmcsd/mmcsd_sdio.c:3363]:
```suggestion
              finfo("Capacity: %lu blocks\n", (unsigned long)priv->nblocks);
```

**Review by pkarashchenko — APPROVED** [review]

File: drivers/mmcsd/mmcsd_sdio.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/mmcsd/mmcsd_sdio.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

File: drivers/mmcsd/mmcsd_sdio.c [hunk 2]
```diff
@@ -3377,8 +3359,7 @@ static int mmcsd_probe(FAR struct mmcsd_state_s *priv)
             {
               /* Yes...  */
 
-              finfo("Capacity: %lu Kbytes\n",
-                    (unsigned long)(priv->capacity / 1024));
+              finfo("Capacity: %lu blocks\n",  (unsigned long)priv->nblocks);
```
Comment (acassis) [inline drivers/mmcsd/mmcsd_sdio.c:3362]: I don't think this is a good change! You are replacing human readable Kbytes by number of blocks. It makes user life more complicated. Please keep the Capacity in Kbytes/Mbytes.

**Review by hartmannathan — COMMENTED** [review]
If the goal is to remove the many checks for `#ifdef CONFIG_HAVE_LONG_LONG` then maybe another approach is to add:

```
#ifdef CONFIG_HAVE_LONG_LONG
typedef uint64_t mmcsd_capacity_t;
#else
typedef uint64_t mmcsd_capacity_t;
#endif
```

and change the typecasts in different places to mmcsd_capacity_t?

File: drivers/mmcsd/mmcsd_sdio.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/mmcsd/mmcsd_sdio.c]: Ok, the capacity bring back again.

**Review by acassis — APPROVED** [review]

File: drivers/mmcsd/mmcsd_sdio.c [hunk 3]
```diff
@@ -692,14 +682,10 @@ static void mmcsd_decode_csd(FAR struct mmcsd_state_s *priv, uint32_t csd[4])
...
-#ifdef CONFIG_HAVE_LONG_LONG
-          priv->capacity        = ((uint64_t)(csize + 1)) << 19;
-#else
-          priv->capacity        = (csize + 1) << 19;
-#endif
+
           priv->blockshift      = 9;
           priv->blocksize       = 1 << 9;
-          priv->nblocks         = priv->capacity >> 9;
+          priv->nblocks         = (csize + 1) << (19 - 9);
```
Comment (pkarashchenko) [inline drivers/mmcsd/mmcsd_sdio.c:688]: Is it reasonable to use blockshift value here and above instead of duplicating 9? Or those are different 9s?

**Review by pkarashchenko — APPROVED** [review]

File: drivers/mmcsd/mmcsd_sdio.c (hunk 3 above)
Comment (xiaoxiang781216) [reply drivers/mmcsd/mmcsd_sdio.c]: Done

**Pull request merged**